engine/execution/pruner/core.go

```diff
@@ -155,3 +164,18 @@ func makeBlockIteratorCreator(
 		return next, header.Height, nil
 	}, nil
 }
+
+// estimateBatchProcessing estimates the number of batches and the total duration
+func EstimateBatchProcessing(
+	start, end uint64, batchSize uint, sleepAfterEachBatchCommit time.Duration, commitDuration time.Duration) (
+	batchCount uint64, totalDuration time.Duration) {
```

Comment:
if you're going to split it only a new line, I think it should use standard formatting.
```suggestion
	start, end uint64,
	batchSize uint,
	sleepAfterEachBatchCommit time.Duration,
	commitDuration time.Duration,
) (batchCount uint64, totalDuration time.Duration) {
```